raiden/connection_manager.py

```diff
@@ -0,0 +1,259 @@
+        self,
+        funds,
+        initial_channel_target=3,
+        joinable_funds_target=.4
+    ):
+        """Connect to the network.
+        Use this to establish a connection with the token network.
+        Args:
+            funds (int): the amount of tokens spendable for this
+            ConnectionManager.
+            initial_channel_target (int): number of channels to open immediately
+            joinable_funds_target (float): amount of funds not initially assigned
+        """
+        if funds <= 0:
+            raise ValueError('connecting needs a positive value for `funds`')
+
+        self.initial_channel_target = initial_channel_target
+        self.joinable_funds_target = joinable_funds_target
+
+        if len(self.channelgraph.graph.nodes()) == 0:
+            with self.lock:
+                log.debug('bootstrapping token network.')
+                # make ourselves visible
+                self.raiden.api.open(
+                    self.token_address,
+                    ConnectionManager.BOOTSTRAP_ADDR
+                )
+
+        with self.lock:
+            self.funds = funds
+            funding = self.initial_funding_per_partner
+            for partner in self.find_new_partners(self.initial_channel_target):
+                self.raiden.api.open(
+                    self.token_address,
+                    partner,
+                )
+                self.raiden.api.deposit(
+                    self.token_address,
+                    partner,
+                    funding
+                )
+
+    def leave(self, wait_for_settle=True, timeout=30):
+        """
+        Leave the token network.
+        This implies closing all open channels and optionally wait for
+        settlement.
+        Args:
+            wait_for_settle (bool): block until successful settlement?
+            timeout (float): maximum time to wait
+        """
+        with self.lock:
+            self.initial_channel_target = 0
+            open_channels = self.open_channels
+            channel_specs = [(
+                self.token_address,
+                c.partner_address) for c in open_channels]
+            for channel in channel_specs:
+                try:
+                    self.raiden.api.close(*channel),
+                except RuntimeError:
+                    # if the error wasn't that the channel was already closed: raise
+                    if channel[1] in [c.partner_address for c in self.open_channels]:
+                        raise
+
+            # force state update
+            self.raiden.poll_blockchain_events(self.raiden.get_block_number())
+
+            if wait_for_settle:
+                try:
+                    with gevent.timeout.Timeout(timeout):
+                        while any(c.state != CHANNEL_STATE_SETTLED for c in open_channels):
+                            # force state update
+                            self.raiden.poll_blockchain_events(self.raiden.get_block_number())
```

Comment:
ah thanks!
could we rename the to into
- raiden.get_last_block_number
- chain.poll_current_block_number
?